### PR TITLE
Upgrade vaadin and importmap and add caching

### PR DESCRIPTION
**Pull request by phillip-kruger**
This PR upgrade to the latest vaadin, that is now build with a new version of mvnpm, that allows caching of these resources.

### CODE REVIEW

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/devui/runtime/MvnpmHandler.java [hunk 1]
```diff
@@ -37,20 +44,38 @@ public void handle(RoutingContext event) {
...
         event.next();
     }
 
+    private String formatDate(long m) {
+        Instant i = Instant.ofEpochMilli(m);
+        return formatDate(i);
+    }
+
+    private String formatDate(TemporalAccessor t) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("EEE, dd MMM yyyy HH:mm:ss z", Locale.ENGLISH)
```
Comment (cescoffier) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/devui/runtime/MvnpmHandler.java:74]: Isn't that a bit expensive? 
Isn't that the API that was made thread-safe? 

@geoand do you know?

**Review by cescoffier — APPROVED** [review]
LGTM

One comment about the data formatter.

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/devui/runtime/MvnpmHandler.java (hunk 1 above)
Comment (geoand) [reply extensions/vertx-http/runtime/src/main/java/io/quarkus/devui/runtime/MvnpmHandler.java:74]: > Isn't that the API that was made thread-safe

Correct. 

> Isn't that a bit expensive?

Yeah, it's not very performant AFAIK. Netty has `DateFormatter` but I don't know how it compares and if it has been optimized. @franz1981 do you know?

Comment (franz1981) [reply extensions/vertx-http/runtime/src/main/java/io/quarkus/devui/runtime/MvnpmHandler.java:74]: If it's an hot path probably worth using a whole different pattern...
The easier thing is to save in a static final field the formatter, which is immutable and thread safe (please check the doc yourself, I am still on PTO so I have limited GitHub access..).

The best would be to have a volatile field with the formatted string, updated using a timer (or some scheduled task) with second granularity. It's less costly and in the hot path the best solution overall.

Comment (franz1981) [reply extensions/vertx-http/runtime/src/main/java/io/quarkus/devui/runtime/MvnpmHandler.java:74]: As usual, measure, not guess; check if the date formatting is a costly hot path and you have few options to reduce its overhead...
@geoand I didn't checked Netty yes will do with my beloved desktop at hand 👍

Comment (phillip-kruger) [reply extensions/vertx-http/runtime/src/main/java/io/quarkus/devui/runtime/MvnpmHandler.java:74]: This is Dev mode only, and also Dev UI only, so for now I think we can merge.

**Pull request merged**